packages/stripe_web/lib/src/widgets/payment_element.dart

```diff
@@ -42,11 +54,17 @@ class PaymentElement extends StatefulWidget {
   final js.PaymentElementOptionsTerms? terms;
   final js.PaymentElementWalletOptions? wallets;
   final js.PaymentElementApplePayOptions? applePay;
+  final String? locale;
 
   const PaymentElement({
     super.key,
-    required this.clientSecret,
+    this.clientSecret,
```

Comment:
I am thinking maybe we should split the widget into two constructors to make it clear for consumer what to add. In case for:

[without intent](https://docs.stripe.com/js/elements_object/create_without_intent ) => mode, anount and currency are required

[with intent](https://docs.stripe.com/js/elements_object/create) => client secret is required.